TrackletAlgorithm/TrackletProcessor.h

```diff
@@ -358,7 +358,9 @@ ::rinv * const rinv, TrackletParameters::PHI0PAR * const phi0, TP::Types::z0 * c
   const ap_int<TrackletParameters::kTParPhi0Size + 2> phicrit = *phi0 - (*rinv>>8)*ifactor;
   const bool keep = (phicrit > phicritmincut) && (phicrit < phicritmaxcut);
 
-  return valid_rinv && valid_z0 && keep;
+  bool valid_tmax=std::abs(*t)<4000; //FIXME this is a protection but need to implement consistent with C++ emu.
```

Comment:
@aryd Where does the 4000 come from? I'm guessing it's not a magic number in the emulation, so could it be derived from meaningfully named constants here?

I'm also not sure I understand the comment. Does the "FIXME" imply that this should be removed from the emulation and HLS at some point?